cmake/os-detection.cmake

```diff
@@ -7,6 +7,9 @@ function (get_os_id OS_ID)
   if (EXISTS "/etc/os-release")
     read_key("ID" _os_id)
   endif()
+  if (_os_id STREQUAL "opensuse-leap")
+    set(_os_id "sles")
```

Comment:
We need some support added to rocm_cmake.

Reply:
https://github.com/RadeonOpenCompute/rocm-cmake/pull/28 Forgot to link this in the PR